doc/dev/deduplication.rst

```diff
@@ -149,3 +149,64 @@ Aside from radosgw, completing work on manifest object support in the
 OSD particularly as it relates to snapshots would be the next step for
 rbd and cephfs workloads.
 
+How to use deduplication
+========================
+
+Currently, Ceph provides deduplication using RADOS machinery.
+The below is explanation about how users use deduplication.
+
+
+1. Find dedupable chunks using ceph-dedup-tool. ::
+
+    ceph-dedup-tool --op estimate --pool $POOL --chunk-size chunk_size  
+      --chunk-algorithm fixed|fastcdc --fingerprint-algorithm sha1|sha256|sha512
+
+  This CLI command will show how many storage space can be saved when deduplication
+  is applied on the pool. If the amount of the saved space is higher than user's expectation,
+  the pool probably is worth performing deduplication. 
+  Users should specify $POOL where the object---the users want to perform
+  deduplication---is stored. The users also need to run ceph-dedup-tool multiple time
+  with varying ``chunk_size`` to find the optimal chunk size. Note that the
+  optimal value probably differs in the content of each object in case of fastcdc
+  chunk algorithm (not fixed).
+
+
+2. Run dedup command (there are two ways). ::
+  
+    ceph-dedup-tool --op object-dedup --pool POOL --object OID --chunk-pool POOL
+      --fingerprint-algorithm FP --dedup-cdc-chunk-size CHUNK_SIZE
+
+  object-dedup command triggers to perform deduplication on the object specified by OID.
+  Users need to use all parameter above---FP and CHUNK_SIZE---based on the result from step 1.
+  After this command is done, note that fastcdc will be set by default and other parameters 
+  such as FP and CHUNK_SIZE also will be set as the POOL's default parameter.
+  Deduplicated objects will appear in the chunk pool.
+
```

Comment:
Formatting and grammar suggestions:
>The \`\`object-dedup\`\` command triggers deduplication on the RADOS object specified by \`\`OID\`\`.
>All parameters shown above must be specified.  \`\`FP\`\` and \`\`CHUNK_SIZE\`\` should be taken from
>the results of step 1 above..
>Note that when this command is executed, \`\`fastcdc\`\` will be set by default and other parameters 
  such as \`\`FP\`\` and \`\`CHUNK_SIZE\`\` will be set as defaults for the pool.
  Deduplicated objects will appear in the chunk pool.